src/gluonts/model/deepar/_network.py

```diff
@@ -416,10 +416,13 @@ def hybrid_forward(
             else observed_values.min(axis=-1, keepdims=False)
         )
 
-        weighted_loss = weighted_average(
+        weighted_loss = masked_weighted_average(
             F=F, x=loss, weights=loss_weights, axis=1
         )
 
+        # need to mask possible nans and -inf
+        loss = F.where(condition=loss_weights, x=loss, y=F.zeros_like(loss))
```

Comment:
Why masking again? Isn’t the weighted average supposed to be doing this already?

Reply:
This is just loss, not weigthed loss. Idk why there are two here, however, this is how it was, I just changed it so it gets properly masked here too.

Reply:
Hmmm I see. Yeah I’m not sure why there are two here. Anyway I would leave it as it is, without doing the masking in this case. This second output should not be used (in my opinion it shouldn’t be exposed at all) and it’s probably not used by the default training loop, for example.

Reply:
Ok, but then it doesn't make any sense not to change it? The main point of this PR is to mask the loss properly. So whatever is done with that loss, it cannot be done properly with '-inf's in there...

Reply:
ok, sounds good